pkg/virtctl/vnc/vnc.go

```diff
@@ -164,76 +167,84 @@ func (o *VNC) Run(cmd *cobra.Command, args []string) error {
 		listenResChan <- err
 	}()
 
-	// execute VNC
-	go func() {
-		defer close(doneChan)
-		port := ln.Addr().(*net.TCPAddr).Port
-		args := []string{}
-
-		vncBin := ""
-		osType := runtime.GOOS
-		switch osType {
-		case "darwin":
-			if matches, err := filepath.Glob(TIGER_VNC_PATTERN); err == nil && len(matches) > 0 {
-				// Always use the latest version
-				vncBin = matches[len(matches)-1]
-				args = tigerVncArgs(port)
-			} else if err == filepath.ErrBadPattern {
-				viewResChan <- err
-				return
-			} else if _, err := os.Stat(CHICKEN_VNC); err == nil {
-				vncBin = CHICKEN_VNC
-				args = chickenVncArgs(port)
-			} else if !os.IsNotExist(err) {
-				viewResChan <- err
-				return
-			} else if _, err := os.Stat(REAL_VNC); err == nil {
-				vncBin = REAL_VNC
-				args = realVncArgs(port)
-			} else if !os.IsNotExist(err) {
-				viewResChan <- err
-				return
-			} else if _, err := exec.LookPath(REMOTE_VIEWER); err == nil {
-				// fall back to user supplied script/binary in path
+	port := ln.Addr().(*net.TCPAddr).Port
+
+	if onlyProxy {
+		fmt.Printf("VNC Proxy localhost port: %d\n", port)
+	}
+
+	// If onlyProxy flag is passed, do not execute VNC
+	if !onlyProxy {
+		// execute VNC
+		go func() {
```

Comment:
Could you extract this goroutine to a new function? It would improve readability.